firmware/ec/src/devices/powerSource.c

```diff
@@ -275,56 +275,82 @@ pwr_process_get_status_parameters_data(ePower_StatusParamId paramIndex,
     switch (paramIndex) {
         case PWR_STAT_POE_AVAILABILITY: {
             if ((Power_SourceInfo[PWR_SRC_POE].state == PWR_SRC_ACTIVE) ||
-                (Power_SourceInfo[PWR_SRC_POE].state == PWR_SRC_AVAILABLE))
+                (Power_SourceInfo[PWR_SRC_POE].state == PWR_SRC_AVAILABLE)) {
```

Comment:
Lets init *pPowerStatusData = 0;  before switch case . . . . 
No need for the same code . . . .